cis194/week6/stevemao/Fibonacci.hs

```diff
@@ -0,0 +1,55 @@
+{-# OPTIONS_GHC -Wall #-}
+module Fibonacci where
+
+fib :: Integer -> Integer
+fib 0 = 0
+fib 1 = 1
+fib n = (fib $ n - 1) + (fib $ n - 2)
+
+fibs1 :: [Integer]
+fibs1 = fmap fib [0..]
+
+fib' :: Integer -> Integer
+fib' 0 = 0
+fib' 1 = 1
+fib' n = snd $ foldr (\_ acc -> (snd acc, fst acc + snd acc)) (0, 1) [2..n]
```

Comment:
Question: in order to generating `0, 1, 1, 2, 3` that is the first 5 numbers. How many times is addition invoked?

Reply:
twice?

Reply:
I think it's going to be more than 3 times.

Reply:
I still don't see duplicated additions. It should just do the following ones:
```haskell
0 + 1
1 + 1
2 + 1
```

Reply:
To get to the fibonacci number at position n, yes. 
How about using `fib'` to generate the fibonacci numbers from 0 to the number at position n.